docs/release-notes/changelog/v5.0.x.rst

```diff
@@ -4,6 +4,37 @@ Open MPI v5.0.x series
 This file contains all the NEWS updates for the Open MPI v5.0.x
 series, in reverse chronological order.
 
+Open MPI Version v5.0.6.rc1
+--------------------------
+:Date: 7 November 2024
+
+- Internal PMIx and PRRTe version:
+
+    - PMIx (v5.0.4). Commit hash: ``c6b895e261384016e154865dd236faaaa471d173``.
+    - PRRTE (v3.0.7). Commit hash: ``1b6e6d7e3cb8c0837eba1730cb62b62186ab042c``.
+
+    Session Management Enhancements:
+    -Finalization Control: Ensures proper finalization of classes when references to the OPAL utility layer reach zero.
+    A new smoke test has been added to validate this behavior.
+    -Test and CI Additions: A smoke test was introduced for session management, also integrated into macOS GitHub CI to ensure stability across platforms.
+    -Multiple Init/Finalize Handling: Improvements to handling session multiple initializations and finalizations, addressing issues with parameter management and session destruction.
+
```

Comment:
Looks fundamentally good, wondering whether we should add in the `Session Management Enhancements` also a line about the fix to Waitall/Waitany/Testall/Testany regarding the recognizing requests from the same instance. (see PR https://github.com/open-mpi/ompi/pull/12711)